### ISSUE TITLE
Please improve access the ApplicationSettings in out of proc azure functions.

### ISSUE THREAD

**Issue by sonic1981** [Azure/azure-functions-dotnet-worker#782]
#### What problem would the feature you're requesting solve? Please describe.
I need to be able to access some values that another service are adding to `ApplicationSettings` on an Azure Service Bus. There is no simple way to do this in an out of proc service. I have (after much digging discovered the following code will allow me to access this information:

```c#
IReadOnlyDictionary<string, object> bindingData = context.BindingContext.BindingData;

if (bindingData.ContainsKey("UserProperties") == false)
{
    throw new Exception("Service bus message is missing UserProperties binding data");
}

string userPropertiesStr = bindingData["UserProperties"].ToString();
if (string.IsNullOrEmpty(userPropertiesStr))
{
    throw new Exception("UserProperties is null or empty");
}

JsonDocument json = JsonDocument.Parse(userPropertiesStr);
JsonElement xProp = json.RootElement.GetProperty("x");
string x = serializationProp.GetString();
```
[See my SO question](https://stackoverflow.com/questions/70709823/how-can-i-access-the-applicationproperties-of-a-service-bus-message-in-a-azure-f)

But really, this should be a LOT easier?
#### Describe the solution you'd like
why not just expose a dictionary of `ApplicationSettings` in the `FunctionContext`? Then that code could be simplified to:

```c#
string x = context.ApplicationSettings["x"];
```

#### Describe alternatives you've considered
[See my SO question](https://stackoverflow.com/questions/70709823/how-can-i-access-the-applicationproperties-of-a-service-bus-message-in-a-azure-f)

**Comment by brettsam** [comment]
Agreed. We need to make some improvements here -- we'll need to do some design here for a clean solution.

**Comment by paul-datatech911** [comment]
FWIW - After discussion on StackOverflow about this - in particular ServiceBus Trigger/Bindings -  the answer for me was to write my own generic extensions that take the string content & FunctionContext and parse out our version of a Message/BrokeredMessage that includes built-in System and UserProperties and a ParsedObject<T>.   Not ideal but easy enough until the Functions Worker team implements extensions that do this (although it seemed based on SO discussion that wouldn't happen).

**Comment by brettsam** [comment]
This was implemented in #944

**Comment by danejur** [comment]
@brettsam I'm not sure I'm seeing how this issue was resolved in #944. It looks like that PR was fairly specific to Application Insights, whereas this issue is related to the ApplicationSettings JSON blob that is returned as part of the `FunctionContext` (and specifically how it is annoying to access the properties from said JSON).

Perhaps this was prematurely closed? I looked through the PR, but saw nothing related to this issue.

**Comment by meta-space** [comment]
> @brettsam I'm not sure I'm seeing how this issue was resolved in #944. It looks like that PR was fairly specific to Application Insights, whereas this issue is related to the ApplicationSettings JSON blob that is returned as part of the `FunctionContext` (and specifically how it is annoying to access the properties from said JSON).
> 
> Perhaps this was prematurely closed? I looked through the PR, but saw nothing related to this issue.

Agreed. This is definitely still an unsolved issue as of Sept. 2023
